docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/waiting/ClusterHealthCheck.java

```diff
@@ -37,5 +43,26 @@ static ClusterHealthCheck serviceHealthCheck(String containerName, HealthCheck<C
         };
     }
 
-    SuccessOrFailure isClusterHealthy(Cluster cluster);
+    static ClusterHealthCheck nativeHealthChecks() {
+        return cluster -> {
+            Set<String> unhealthyContainers = new LinkedHashSet<>();
```

Comment:
Personally I'd kinda prefer to Java8 this and do a stream, map thing - it emphasises there's no mutation or anything fancy happening here.